lib/core/wallet_loading_service.dart

```diff
@@ -38,15 +39,43 @@ class WalletLoadingService {
   }
 
   Future<WalletBase> load(WalletType type, String name) async {
-    final walletService = walletServiceFactory.call(type);
-    final password = await keyService.getWalletPassword(walletName: name);
-    final wallet = await walletService.openWallet(name, password);
+    try {
+      final walletService = walletServiceFactory.call(type);
+      final password = await keyService.getWalletPassword(walletName: name);
+      final wallet = await walletService.openWallet(name, password);
 
-    if (type == WalletType.monero) {
-      await updateMoneroWalletPassword(wallet);
-    }
+      if (type == WalletType.monero) {
+        await updateMoneroWalletPassword(wallet);
+      }
+
+      return wallet;
+    } catch (error, stack) {
+      ExceptionHandler.onError(FlutterErrorDetails(exception: error, stack: stack));
+
+      // try opening another wallet that is not corrupted to give user access to the app
+      final walletInfoSource = await CakeHive.openBox<WalletInfo>(WalletInfo.boxName);
 
-    return wallet;
+      for (var walletInfo in walletInfoSource.values) {
+        try {
+          final walletService = walletServiceFactory.call(walletInfo.type);
+          final password = await keyService.getWalletPassword(walletName: walletInfo.name);
+          final wallet = await walletService.openWallet(walletInfo.name, password);
+
+          if (walletInfo.type == WalletType.monero) {
+            await updateMoneroWalletPassword(wallet);
+          }
+
+          await sharedPreferences.setString(PreferencesKey.currentWalletName, wallet.name);
+          await sharedPreferences.setInt(
+              PreferencesKey.currentWalletType, serializeToInt(wallet.type));
```

Comment:
Is setting the wallet as current here necessary? It could have unintended side-effects during background sync. All the other calls to the `WalletLoadingService.load` method run `AppStore.changeCurrentWallet` which sets these shared preferences using a mobx-reaction defined in `lib/reactions/on_current_wallet_change.dart`.

Reply:
if I understand your concern correctly, then it should have no side-effect, as it will return the wallet that is not corrupted anyway, so `AppStore.changeCurrentWallet` will be called with the not corrupted wallet, so it will just re-assign those preferences values again with the same values, I just added it to be on the safe side.

feel free to test it tho.

I tried testing it by specifically making it throw an exception when loading the current wallet while having multiple wallets, and it successfully switched to the second wallet I had